src/data/vector2.lean

```diff
@@ -169,6 +189,26 @@ lemma insert_nth_comm (a b : α) (i j : fin (n+1)) (h : i ≤ j) :
 
 end insert_nth
 
+section update_nth
+
+def update_nth (v : vector α n) (i : fin n) (a : α) : vector α n :=
+⟨v.1.update_nth i.1 a, by rw [list.update_nth_length, v.2]⟩
+
+@[simp] lemma nth_update_nth_eq (v : vector α n) (i : fin n) (a : α) :
+  (v.update_nth i a).nth i = a :=
+by cases v; cases i; simp [vector.update_nth, vector.nth_eq_nth_le]
+
+lemma nth_update_nth_ne {v : vector α n} {i j : fin n} (h : i ≠ j) (a : α) :
+  (v.update_nth i a).nth j = v.nth j :=
+by cases v; cases i; cases j; simp [vector.update_nth, vector.nth_eq_nth_le,
+  list.nth_le_update_nth_of_ne (fin.vne_of_ne h)]
+
+lemma nth_update_nth {v : vector α n} {i j : fin n} (a : α) :
+  (v.update_nth i a).nth j = if i = j then a else v.nth j :=
```

Comment:
The meaning of this name and `nth_update_nth_eq` are indistinguishable. Could you do at least one of the following renamings:
* `nth_update_nth_eq` -> `nth_update_nth[_eq]_same` (and also for list)
* `nth_update_nth` -> `nth_update_nth_eq_if`

Reply:
I think `nth_update_nth_eq_ite` is better than `nth_update_nth_eq_if`.

Reply:
I thought we called this operation `if`. For example, `if_pos` and `if_neg` exist, but `ite_pos` and `ite_neg` don't.

Reply:
You're right. My mistake